test/parallel/test-cluster-debug-port.js

```diff
@@ -1,42 +1,69 @@
 'use strict';
+
+// this is hack to make master process think it is running with --debug
+if (process.execArgv.length == 0) {
+  process.execArgv = ['--debug'];
+}
+
 require('../common');
 const assert = require('assert');
 const cluster = require('cluster');
 
 if (cluster.isMaster) {
-
   function checkExitCode(code, signal) {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
   }
 
-  console.log('forked worker should not have --debug-port');
-  cluster.fork().on('exit', checkExitCode);
+  console.log('if master is debugging with default port, ' +
+    'forked worker should have debugPort, with offset = 1');
+  cluster.fork({
+    portSet: process.debugPort + 1
+  }).on('exit', checkExitCode);
+
+  console.log('for next forked workers offset should increment');
+  cluster.fork({
+    portSet: process.debugPort + 2
+  }).on('exit', checkExitCode);
 
+  console.log('if master is debugging on non-default port ' +
+    'worker ports should be incremented from it');
   cluster.setupMaster({
-    execArgv: ['--debug-port=' + process.debugPort]
+    execArgv: ['--debug=12000']
   });
+  cluster.fork({
+    portSet: 12000 + 3
+  }).on('exit', checkExitCode);
 
-  console.log('forked worker should have --debug-port, with offset = 1');
+  console.log('debugging with --debug=hostname:port format should also work');
+  cluster.setupMaster({
+    execArgv: ['--debug=localhost:12100']
+  });
   cluster.fork({
-    portSet: process.debugPort + 1
+    portSet: 12100 + 4
   }).on('exit', checkExitCode);
 
+  console.log('debugging with --debug=ipv4addr:port format should also work');
   cluster.setupMaster({
-    execArgv: [`--debug-port=${process.debugPort}`,
-               `--debug=${process.debugPort}`]
+    execArgv: ['--debug=127.0.0.1:12200']
   });
+  cluster.fork({
+    portSet: 12200 + 5
+  }).on('exit', checkExitCode);
 
-  console.log('forked worker should have --debug-port, with offset = 2');
+  console.log('debugging with --debug=[ipv6addr]:port format should also work');
```

Comment:
Yes, there are ipv6-only tests that bail immediately if it's not available. Besides, it's easy enough to support with just an added conditional.